src/common/Preforker.h

```diff
@@ -58,6 +58,7 @@ class Preforker {
       ::close(fd[0]);
     } else {
       ::close(fd[1]);
+      signal(SIGHUP, SIG_IGN);
```

Comment:
I think this needs to go before the fork? Should not the child also ignore SIGHUP until its handlers are configured?

Reply:
If the child receive the SIGHUP the parent won't hit this call trace issue. The pipe read() in parent will just return 0 as the EOF and then the parent will exist normally.

But maybe we should ignore the SIGHUP signal in child process before the child daemon setup its own handler for the logrotation.

Reply:
@lxbsz  Yes, it's will not produce one Uninterruptible Sleep process  in current submission.  But do you think it‘s a better choice to move the signal response function to before the fork in forker.prefork()?

Reply:
Yeah, that will be perfect.

As @batrick mentioned above, in the case the logrotation service invokes the killall SIGHUP while the ceph's daemon booting and if the daemon's child process here catch the signal it will exit directly, but we should ignore the SIGHUP before its own SIGHUP handler is configured.

Reply:
@lxbsz  @batrick  I have modified it. Please have a review, thank you.

Reply:
This looks good to me.

@batrick  BTW, or should we move this as earlier as possible in the daemon main(), to make sure the daemon won't be killed by the logrotate service ?

Reply:
> @batrick BTW, or should we move this as earlier as possible in the daemon main(), to make sure the daemon won't be killed by the logrotate service ?

That's a race not worth closing IMO.

Reply:
It's also worth noting that there was a race between fork and when `signal(SIGHUP, SIG_IGN)` was called. That's the main reason to move it before the `fork`.